### PR TITLE
Instrument list ordering

### PR DESCRIPTION
**Pull request by tedstrauss**
This allows explicit ordering of subgroups and instruments on the test battery page (instrument_list module).

screenshot from 2016-01-21 10 16 24

### CODE REVIEW

**Comment by driusan** [comment]
I don't understand this ordering.

**Comment by tedstrauss** [comment]
phpcs is passing locally, but not in travis. 
this is an outrage.
who is to blame?

**Comment by driusan** [comment]
Did you include --standard=docs/LorisCS.xml when running it locally? Are you running the same version of phpcs (the one in vendor/bin/ after doing a composer install)?

File: modules/instrument_list/templates/menu_instrument_list.tpl [hunk 1]
```diff
@@ -25,8 +25,8 @@
 	<tbody>
 	   	<tr{if $instruments[group][instrument].isDirectEntry} class="directentry"{/if}>
 	    	<td>
-		    	<a href="{$instruments[group][instrument].testName}/?commentID={$instruments[group][instrument].commentID}">
-	            {$instruments[group][instrument].fullName}</a></td>
+                <a href="main.php?test_name={$instruments[group][instrument].testName}&candID={$candID}&sessionID={$sessionID}&commentID={$instruments[group][instrument].commentID}">
```
Comment (gluneau) [inline modules/instrument_list/templates/menu_instrument_list.tpl]: Let's remove references to main.php

File: php/libraries/Utility.class.inc [hunk 2]
```diff
@@ -894,6 +894,39 @@ class Utility
...
+     * @param string $CandID candidate ID
+     *
+     * @return array
+     * @throws DatabaseException
+     */
+    static function getStageUsingCandID($CandID)
+    {
+        $db    =& Database::singleton();
+        $query = "select DISTINCT Current_stage from session where CandID='$CandID'";
+        $stage = $db->pselect($query);
```
Comment (gluneau) [inline php/libraries/Utility.class.inc]: Please use the prepared statement syntax

File: SQL/Archive/16.04/2016-01-20-InstrumentListOrdering.sql [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# add ordering column (tinyint) to battery
+ALTER TABLE test_battery ADD instr_order tinyint after firstVisit;
+# add ordering column (tinyint) to test_subgroups
+ALTER TABLE test_subgroups ADD group_order tinyint after Subgroup_name;
```
Comment (xlecours) [inline SQL/Archive/16.04/2016-01-20-InstrumentListOrdering.sql]: This should be reflected in the schema. And a patch should be available somewhere else than in the archives.

**Pull request merged**